universum/lib/gravity.py

```diff
@@ -7,28 +9,30 @@
 
 
 class ModuleSettings:
-    def __init__(self, cls, main_settings):
+    def __init__(self, cls: Type['Module'], main_settings: 'ModuleSettings') -> None:
         object.__setattr__(self, "cls", cls)
         object.__setattr__(self, "main_settings", main_settings)
+        # TODO: can't clarify that active_modules[Type[X]] returns X (https://github.com/python/mypy/issues/4928)
+        self.active_modules: Optional[Dict[Type[Module], Module]]
```

Comment:
for some reason mypy lacks support of TypeVar in dicts. I can either leave it as is or refactor in a separate class with its own getter and setter. However that seems excessive, since 'construct_component' already enforces strict type adherence for return value, and 'active_modules' is not used outside of 'gravity' module.

Reply:
As an option, you can turn it into a pair of typed functions.

Reply:
yes, that's similar to the option 1 from the comment above. Still, I don't think it adds much value considering the 'construct_component'

Reply:
I agree.